Add map type property (standard, satellite, hybrid) to MapView.

adamkrell opened the pull request:
Here is an example project demonstrating this pull request: [AKMapView](https://github.com/adamkrell/AKMapView)

paramaggarwal commented:
I suggest using `enum`s instead of `NSUInteger`. Take a look at how the `Image` component [RCTNetworkImageViewManager.m](https://github.com/facebook/react-native/blob/master/Libraries/Image/RCTNetworkImageViewManager.m) uses the `UIViewContentMode` enum with help from [RCTConvert.h](https://github.com/facebook/react-native/blob/master/React/Base/RCTConvert.h).

adamkrell commented:
@paramaggarwal Thanks for pointing me in the right direction. I updated the PR to use an enum for the mapType property. I also updated my example project: [AKMapView](https://github.com/adamkrell/AKMapView)

paramaggarwal commented on React/Views/RCTMapManager.m:
```diff
@@ -41,6 +41,7 @@ - (UIView *)view
 RCT_EXPORT_VIEW_PROPERTY(maxDelta, CGFloat)
 RCT_EXPORT_VIEW_PROPERTY(minDelta, CGFloat)
 RCT_EXPORT_VIEW_PROPERTY(legalLabelInsets, UIEdgeInsets)
+RCT_REMAP_VIEW_PROPERTY(mapType, mapType, MKMapType)
```
You can leave this at `RCT_EXPORT_VIEW_PROPERTY(mapType, MKMapType)` as the implementation of `RCT_EXPORT_VIEW_PROPERTY` is anyways like this:

```objective-c
#define RCT_EXPORT_VIEW_PROPERTY(name, type) RCT_REMAP_VIEW_PROPERTY(name, name, type)
```

adamkrell replied:
@paramaggarwal I didn't realize that. Updated to use ` RCT_EXPORT_VIEW_PROPERTY`.

brentvatne commented on React/Views/RCTMap.m:
```diff
@@ -120,4 +120,8 @@ - (void)setAnnotations:(MKShapeArray *)annotations
 
+- (void)setMapType:(MKMapType)mapType
+{
+    [super setMapType:mapType];
```
nit: this should be indented two spaces

brentvatne commented:
@adamkrell - looks good aside from the one small style nit, can you fix that and squash into one commit?

adamkrell commented:
@brentvatne Fixed coding convention issue and squashed commits.

brentvatne commented:
Looks good to me!

brentvatne commented:
@tadeuzagallo - assigned you to this as your name appears most frequently in related code, re-assign as you see fit of course :smile:

tadeuzagallo commented on React/Views/RCTMap.m:
```diff
@@ -120,4 +120,8 @@ - (void)setAnnotations:(MKShapeArray *)annotations
 
+- (void)setMapType:(MKMapType)mapType
+{
+  [super setMapType:mapType];
```
If it just calls `super`, it's not needed.

adamkrell commented:
@tadeuzagallo I removed the unnecessary changes to RCTMap.

tadeuzagallo commented:
Looks good now, thanks! :smiley:

@facebook-github-bot import
